### PR TITLE
Refactor application settings

### LINKED ISSUE #985: Refactor application settings
**Is your feature request related to a problem? Please describe.**
PHP and Laravel becomes more strictly typed over time. 
Our global settings store however is not typed, causing issues like #982.
These issue cannot be covered easily using our tests.

**Describe the solution you'd like**
Move to a strictly typed global settings storage.

**Describe alternatives you've considered**
*none*

### PR DESCRIPTION
**Pull request by SamuelWei**
# Fixes #985 

## Type (Highlight the corresponding type)
- Bugfix
- Feature
- Documentation
- **Refactoring** (e.g. Style updates, Test implementation, etc.)
- Other (please describe):

## Checklist
- [x] Code updated to current develop branch head
- [x] Passes CI checks
- [x] Is a part of an issue
- [x] Tests added for the bugfix or newly implemented feature, describe below why if not
- [x] Changelog is updated
- [x] Documentation of code and features exists

## Changes

- Drop old global application settings storage
- Add new global application settings storage
- Add database upgrade command for v2/v3 to v4
- Adjust docs
- Cleanup .env

### CODE REVIEW

**Comment by SamuelWei** [comment]
> As discussed, please consider fixing/adding Postgres compatibility by changing (introduced in `2022_09_27_193354_create_sessions_table.php`):
> 
> ```diff
> -            $table->string('id')->index()->unique();
> +            $table->string('id')->primary();
> ```
> 
> ideally this should be fixed in said migration, too.

These issues will be fixed first in #1079 and later adjusted to this PR

File: .env.ci [hunk 1]
```diff
@@ -34,143 +42,4 @@ MAIL_USERNAME=null
...
-# Enable Shibboleth authentication
-SHIBBOLETH_ENABLED=false
-
-# Enabled locales
-# Comma separated list, e.g. de,en
-# If unset all available locales are enabled (lang + resources/custom/lang)
-#ENABLED_LOCALES=de,en
-
-# Default locale
-DEFAULT_LOCALE=en
```
Comment (SamuelWei) [inline .env.ci:81]: Add to docs

File: .env.ci [hunk 2]
```diff
@@ -34,143 +42,4 @@ MAIL_USERNAME=null
 MAIL_PASSWORD=null
 MAIL_ENCRYPTION=null
 MAIL_FROM_ADDRESS=admin@domain.tld
-MAIL_FROM_NAME="${APP_NAME}"
-
-# Enable local authentication
-LOCAL_AUTH_ENABLED=true
```
Comment (SamuelWei) [inline .env.ci:40]: Add to docs

File: .env.ci [hunk 3]
```diff
@@ -34,143 +42,4 @@ MAIL_USERNAME=null
...
-
-# Default locale
-DEFAULT_LOCALE=en
-
-# Don't forget to rebuild frontend on changing any environment variables starting with VITE_
-
-# Theme (resources/sass/theme/)
-VITE_THEME=default
-
-# Maximum length of the room welcome message
```
Comment (SamuelWei) [inline .env.ci:88]: Add to docs

File: .env.ci [hunk 4]
```diff
@@ -34,143 +42,4 @@ MAIL_USERNAME=null
...
-WELCOME_MESSAGE_LIMIT=500
-# Maximum length of the room name
-ROOM_NAME_LIMIT=50
-# Base time in seconds to automatically reload room page to detect change in room status and settings
-# the value is slightly randomized to prevent simultaneous refreshes
-ROOM_REFRESH_RATE=5
```
Comment (SamuelWei) [inline .env.ci:94]: Add to docs

File: config/settings.php [hunk 5]
```diff
@@ -1,112 +1,111 @@
 <?php
 
 return [
+
+    'defaults' => [
+        'general' => [
+            'name' => env('APP_NAME', 'PILOS'),
```
Comment (danielmachill) [inline config/settings.php:7]: check possibility of changing to DEFAULT_APP_NAME

File: lang/fr/settings.php [hunk 6]
```diff
@@ -98,7 +98,7 @@
             ],
         ],
         'room_limit' => [
```
Comment (danielmachill) [inline lang/fr/settings.php:100]: remove to make it show as not completed for online translation

File: storage/settings.json [hunk 7]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (danielmachill) [inline storage/settings.json:1]: remove file

**Review by danielmachill — COMMENTED** [review]
localisation, tests and upgrade documentation not yet reviewed

File: config/settings.php (hunk 5 above)
Comment (SamuelWei) [reply config/settings.php:7]: Not possible, added note to docs to also change .env file

**Pull request merged**